src/TLuaInterpreter.cpp

```diff
@@ -9512,6 +9512,90 @@ int TLuaInterpreter::getAllRoomUserData( lua_State * L )
     }
 }
 
+// getAllRoomsUserDataKeys( roomID )
+// returns a sorted list of the user data keys for ALL rooms.  Will return
+// an empty table if no user data. This will be useful if the user is not the
+// creator of the data and does not know what has been stored in the user data
+// area!
+int TLuaInterpreter::getAllRoomsUserDataKeys( lua_State * L )
```

Comment:
It should return nil if there is no data - makes more sense to write:

```lua
local keys = getAllRoomsUserDataKeys(roomid)
if keys then
  -- blah
```
instead of adding in funny code like:

```lua
local keys = getAllRoomsUserDataKeys(roomId)
if next(keys) then
  --blah
```